llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -2772,6 +2772,40 @@ bool RISCVTTIImpl::canSplatOperand(Instruction *I, int Operand) const {
   }
 }
 
+bool RISCVTTIImpl::tryToSinkVPSplat(VPIntrinsic *VPI,
+                                    SmallVectorImpl<Use *> &Ops) const {
+  Value *EVL = VPI->getVectorLengthParam();
+  if (!EVL)
+    return false;
+
+  for (auto &Op : VPI->operands()) {
+    auto *I = dyn_cast<Instruction>(Op.get());
+    if (!I || I->getParent() == VPI->getParent() ||
```

Comment:
Why do we have the check for I->getParent() == VPI->getParent()? The non-VP code doesn't seem to need it

Reply:
The caller also has a check for same basic block after this returns. For the non-VP case the pattern to sink is two instructions. 0, 1, or 2 of them may be in the correct basic block. It's easier to let the caller figure it which need to be sunk and where the insertion needs to occur.